### PR TITLE
Add support for custom LearningRateSchedule for TF2

### PR DESCRIPTION
**Pull request by kaixih**
## Checklist before submitting

- [Yes] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [No] Did you update the docs?
- [No] Did you write any tests to validate this change?  
- [No] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description
This PR fixes an issue the current horovod optimizer wrapper cannot correctly deal with the custom learning rate schedule.

fyi. @nluehr @DEKHTIARJonathan 
Fixes # (issue).
Without this PR, the following script would fail:
```python
import tensorflow as tf
import numpy as np
from tensorflow.keras import layers

import horovod.tensorflow as hvd
hvd.init()
gpus = tf.config.experimental.list_physical_devices('GPU')
for gpu in gpus:
  tf.config.experimental.set_memory_growth(gpu, True)
if gpus:
  tf.config.experimental.set_visible_devices(gpus[hvd.local_rank()], 'GPU')

model = tf.keras.Sequential()
model.add(tf.keras.layers.Dense(8))
model.add(tf.keras.layers.Dense(1))

class CustomLearningRateSchedule(
    tf.keras.optimizers.schedules.LearningRateSchedule):
  def __init__(self, lr_values, name):
    super(CustomLearningRateSchedule, self).__init__()
    self.lr_values = lr_values
    self.name = name
    self.learning_rate_ops_cache = {}

  def __call__(self, step):
    if tf.executing_eagerly():
      return self.lr_values * step

    graph = tf.compat.v1.get_default_graph()
    if graph not in self.learning_rate_ops_cache:
      self.learning_rate_ops_cache[graph] = self.lr_values * step
    return self.learning_rate_ops_cache[graph]

  def get_config(self):
    return {
        'lr_values': self.lr_values,
        'name': self.name
    }


learning_rate = CustomLearningRateSchedule(0.001, "toy")
opt = tf.keras.optimizers.SGD(learning_rate)

#opt = tf.optimizers.SGD(0.001)
opt = hvd.DistributedOptimizer(opt)

model.compile(optimizer=opt, loss='mse')
# This builds the model for the first time:

x = np.random.random((100, 3))
y = np.random.random((100,))
model.fit(x, y, batch_size=32, epochs=10)

```

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

**Comment by tgaddair** [comment]
Hey @kaixih, can you rebase your PR from master? We should have fixed these failing PyTorch tests.

**Comment by kaixih** [comment]
It seems all tests get passed. PTAL. @tgaddair

**Review by tgaddair — APPROVED** [review]
LGTM! Thanks for the contribution.
